### PR TITLE
REPO-4636 [MNT-20863] 

### PR DESCRIPTION
**Pull request by elizastan**
Changing cm:name with REST API /nodes/{nodeId} does not update Primary Path

### CODE REVIEW

File: src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 1]
```diff
@@ -138,6 +142,8 @@ public void setup() throws Exception
 
         permissionService = applicationContext.getBean("permissionService", PermissionService.class);
         authorityService = (AuthorityService) applicationContext.getBean("AuthorityService");
+        nodeService = applicationContext.getBean("NodeService", NodeService.class);
+        AuthenticationUtil.pushAuthentication();
```
Comment (killerboot) [inline src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Why is this line required?

File: src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 2]
```diff
@@ -2313,6 +2313,7 @@ else if (! destNodeTypeQName.equals(nodeTypeQName))
 
             try
             {
+                handleUpdateNodeName(props, nodeRef);
```
Comment (killerboot) [inline src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: Can you please change the name of the method to "handleNodeRename", I think it is more descriptive.

File: src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 3]
```diff
@@ -5602,6 +5608,49 @@ private void validateAuditableProperties(Map<String, Object> givenProperties, No
...
+        // update node
+        // /Company Home/User Homes/user<timestamp>/folder_A01
+        folderA.setName("folder_A01");
+        put(URL_NODES, folderA.getId(), toJsonAsStringNonNull(folderA), null, 200);
+        folderAPath = nodeService.getPath(folderANodeRef);
+        pathElement = (Path.ChildAssocElement) folderAPath.last();
+        qNameLocalName = pathElement.getRef().getQName().getLocalName();
+        assertFalse(folderAName.equals(qNameLocalName));
+
+        folderA.setName("folder_A02");
```
Comment (killerboot) [inline src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Please add variables for all strings.

File: src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 4]
```diff
@@ -5602,6 +5608,49 @@ private void validateAuditableProperties(Map<String, Object> givenProperties, No
...
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("cm:name", "folder_A03");
+        folderA.setProperties(properties);
+        put(URL_NODES, folderA.getId(), toJsonAsStringNonNull(folderA), null, 200);
+
+        folderAPath = nodeService.getPath(folderANodeRef);
```
Comment (killerboot) [inline src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Please check the full path.
Also it would be better to check the path of "folder_B" as the rename should impact children's path as well.

Comment (elizastan) [reply src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: I won't do that. I cannot check the full path because it includes the user <timestamp>.

Comment (killerboot) [reply src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: user1 is a username with timestamp and it is a variable, see https://github.com/Alfresco/alfresco-remote-api/blob/master/src/test/java/org/alfresco/rest/api/tests/AbstractBaseApiTest.java#L163

Comment (elizastan) [reply src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Done :)
